scripts/main.py

```diff
@@ -87,6 +100,62 @@ def _get_image_filename(self, file_path):
             .replace(" ", "")
         )
 
+    def _deprecate_previous_if_any(
+        self, latest_issue_url: str, latest_comment_url: str
+    ):
+        """Tell the previous comment about the new one."""
+        deprecation_notice = f"__DEPRECATED__: _This screenshot is no longer up-to-date. The latest version can be found [here]({latest_comment_url})_."
+
+        url_list_comments_in_issue = (
+            f"{self.GITHUB_API_URL}/repos/{self.configuration.GITHUB_REPOSITORY}/"
+            f"issues/{self.configuration.GITHUB_PULL_REQUEST_NUMBER}/comments"
+        )
+
+        response = requests.get(
+            url_list_comments_in_issue, headers=self._request_headers
+        )
+
+        if response.status_code != 200:
+            print_message(
+                "Unable to list previous comments; we will thus not attempt to deprecate any previous comment"
+            )
+            return
+
+        elif comments := response.json():
+            # get the latest 2 comments from a bot if any
+            # FIX ME: What if there are multiple bots posting across Issues / PRs?
+            from_bot_same_issue_or_pr = filter(
+                lambda c: c["user"]["login"] == "github-actions[bot]"
+                and c["issue_url"] == latest_issue_url,
+                comments,
+            )
+            last_comments = list(
+                islice(
+                    sorted(
+                        from_bot_same_issue_or_pr,
+                        key=lambda c: c["created_at"],
+                        reverse=True,
+                    ),
+                    2,
+                )
+            )
+
+            if len(last_comments) != 2:
+                print_message(
+                    "This is the latest commented batch of screenshots, you're all good!"
+                )
+
+            else:
+                edit_past_comment_url = (
+                    f"{self.GITHUB_API_URL}/repos/{self.configuration.GITHUB_REPOSITORY}/"
+                    f"issues/comments/{last_comments[1]['id']}"
+                )
+                response = requests.patch(
+                    edit_past_comment_url,
+                    headers=self._request_headers,
+                    json={"body": deprecation_notice},
+                )
```

Comment:
As said, I think you can just delete the comment, saying that there used to be a comment but not link the screenshot anymore is not useful.

Reply:
My thinking was to allow for comparisons across comments featuring screenshots. Under the current implementation, the user can use the "show edits" on a deprecated comment to render the previous screenshots. Deleting comments would not allow that.